external/kafka/src/test/scala/org/apache/spark/streaming/kafka/KafkaRDDSuite.scala

```diff
@@ -68,6 +68,21 @@ class KafkaRDDSuite extends SparkFunSuite with BeforeAndAfterAll {
 
     val received = rdd.map(_._2).collect.toSet
     assert(received === messages)
+
+    // size-related method optimizations return sane results
+    assert(rdd.count === messages.size)
+    assert(rdd.countApprox(0).getFinalValue.mean === messages.size)
+    assert(! rdd.isEmpty)
+    assert(rdd.take(1).size === 1)
+    assert(messages(rdd.take(1).head._2))
```

Comment:
What does this check? Shouldnt it check that `rdd.take(1) === "the"  // whatever is expected`